nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -0,0 +1,148 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.docker-containers;
+
+  dockerContainer =
+    { name, config, ... }: {
+
+      options = {
+
+        image = mkOption {
+          type = types.str;
+          description = "Docker image to run.";
+          example = "library/hello-world";
+        };
+
+        cmd = mkOption {
+          type =  with types; listOf str;
+          default = [];
+          description = "Commandline arguments to pass to the image's entrypoint.";
+        };
+
+        entrypoint = mkOption {
+          type = with types; nullOr str;
+          description = "Overwrite the default entrypoint of the image.";
+          default = null;
+        };
+
+        environment = mkOption {
+          type = with types; attrsOf str;
+          default = {};
+          description = "Environment variables to set for this container.";
+          example = {
+            DATABASE_HOST = "db.example.com";
+            DATABASE_PORT = "3306";
+          };
+        };
+
+        log-driver = mkOption {
+          type = types.str;
+          default = "none";
+          description = ''
+            Logging driver for the container.  The default of "none" means that
+            the container's logs will be handled as part of the systemd unit.
+            Setting this to "journald" will result in duplicate logging, but
+            the container's logs will be visible to the `docker logs` command.
+          '';
+        };
+
+        ports = mkOption {
+          type = with types; attrsOf str;
+          default = {};
+          description = "Network ports to forward from the host to this container.";
```

Comment:
You should probably indicate that `key = value` indicates [host -> container] / [container -> host], because I don't know what it is just from this

Reply:
I'm kinda leaning toward making this a list of strings, similar to the `volumes` option, because the format is sufficiently complex that it's hard to fully translate to an attribute set without needing another layer of documentation.